src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs

```diff
@@ -44,6 +44,10 @@ public object LaunchProcess(string processPath, string arguments, string working
                     }
                 }
 
+                // VSTEST_WINAPPHOST_DOTNET_ROOT is made to flow by SDK to support apphost and --arch feature
+                // we remove to not pollute host process environment variables
+                process.StartInfo.RemoveEnvironmentVariable("VSTEST_WINAPPHOST_DOTNET_ROOT");
```

Comment:
I don't think this is worth adding another public API to VSTest, users won't try to re-use this variable name inside of their process, and having additional env variable has no impact on performance.